ha/setup/ha_setup.py

```diff
@@ -238,6 +241,38 @@ def _get_s3_instance(self, machine_id: str) -> int:
             Log.error(f"Found {s3_instances} which is invalid s3 instance count. Error: {e}")
             raise HaConfigException(f"Found {s3_instances} which is invalid s3 instance count.")
 
+    def _update_env(self, node_type: str, cluster_type: str) -> None:
+        """
+        Update env like VM, HW
+        """
+        Log.info(f"Detected {node_type} env and cluster_type {cluster_type}.")
+        if "VM" == node_type.upper():
+            Conf.set(const.HA_GLOBAL_INDEX, "CLUSTER_MANAGER.env", node_type.upper())
+        else:
+            # TODO: check if any env available other than vm, hw
+            Conf.set(const.HA_GLOBAL_INDEX, "CLUSTER_MANAGER.env", "HW")
+        Conf.set(const.HA_GLOBAL_INDEX, "CLUSTER_MANAGER.cluster_type", cluster_type)
+        Log.info("CONFIG: Update ha configuration files")
+        Conf.save(const.HA_GLOBAL_INDEX)
+
+    def _update_controller_interface(self) -> None:
+        """
+        Update HA_CLUSTER_SOFTWARE
+
+        Raises:
```

Comment:
Is this a typo?

Reply:
Fixed

it got added by vscode doc string
removed Raises part as not needed